### PR TITLE
🔨 Switch grapher exports to use CF worker

### LINKED ISSUE #4093: Webfonts aren't loading correctly in Grapher SVG thumbnails
Note the headings aren't using Playfair Display:

Image

Image

These are being loaded as SVGs e.g. https://assets.ourworldindata.org/grapher/exports/number-of-nuclear-weapons-tests.svg

Which loads correctly when visited, but when it's embedded in a page, it doesn't work.

We recently [refactored our Cloudflare functions](https://github.com/owid/owid-grapher/pull/4054), which might have something to do with it.

I'm also noticing that we're not preloading Playfair via a `<link>`, but instead requesting it in the CSS:

Image

Whereas we're linking to Lato: 

Image

Though Lato is also not being used in the embed (note the tail of the g)

Embed:
Image

Loaded directly:
Image

### PR DESCRIPTION
**Pull request by ikesau**
Resolves #4093 and #3661

Supersedes #3746

# Changes 
- Switches all references to `/exports/` to use our dynamic thumbnails instead
  - Except the mockSiteRouter view in `/admin/charts`
- Removes code that baked graphers to `/exports/`
  - Some charts in country profiles will no longer load in the NoJS version of the site because their slugs have updated but `formatWordpressPost` isn't checking redirects. We could:
    1. Fix this in the code
    2. Update the country profile templates
    3. Not fix it
- In all thumbnail imgs where applicable, switches the image extension from .svg to .png, to ensure webfonts always show correctly.

## Caveats
- NoJS country profile pages are still a bit broken, as they don't have the correct entity selected, nor do they take redirects into account

### CODE REVIEW

**Comment by ikesau** [comment]
TODO: country profile pages are embedding query params as [Object object]

File: functions/grapher/[slug].ts [hunk 1]
```diff
@@ -153,8 +157,22 @@ async function handleHtmlPageRequest(
...
+                    generateGrapherImageSrcSet(thumbnailUrl)
+                )
+            },
+        })
+        // Grapher and data pages
+        .on(`img.GrapherImage`, {
+            element: (element) => {
+                element.setAttribute("src", thumbnailUrl)
+            },
+        })
```
Comment (marcelgerber) [inline functions/grapher/[slug].ts:175]: This is not quite the way to go. To see why, go to http://staging-site-remove-grapher-exports/grapher/child-mortality?time=1990 and scroll down to the related charts section.

I would suggest doing it like I did in https://github.com/owid/owid-grapher/pull/3873/files: There, I introduced a `[data-owid-populate-url-params]` attr, and only where that is present do we inject the page's query params.

**Review by marcelgerber — CHANGES_REQUESTED** [review]
Nice effort!

I've spotted a few places where this doesn't quite work just yet, though.

**Comment by ikesau** [comment]
Hmm, still running into issues:

- https://remove-grapher-exports.owid.pages.dev/grapher/mpi-vs-consumption
  - Thumbnail 404's for some reason
- https://remove-grapher-exports.owid.pages.dev/grapher/life-expectancy
  - 500s when loading the [850w source](https://remove-grapher-exports.owid.pages.dev/grapher/life-expectancy.png?imWidth=850), but other resolutions work fine
- https://remove-grapher-exports.owid.pages.dev/grapher/population
  - Seems to fail due to there being too many thumbnails to load

Will have to look into it more later 😩

**Comment by marcelgerber** [comment]
> [remove-grapher-exports.owid.pages.dev/grapher/mpi-vs-consumption](https://remove-grapher-exports.owid.pages.dev/grapher/mpi-vs-consumption)
> * The [850w source](https://remove-grapher-exports.owid.pages.dev/grapher/mpi-vs-consumption.png?imWidth=850) 404's for some reason

It seems that this one has been deleted in prod, and so https://grapher-configs.owid.io/v1/config/by-slug-published/mpi-vs-consumption.json (where the worker is trying to fetch the config from) is 404ing. So, just a red herring (and an issue worth bringing up for staging config fetching).

> * The [850w source](https://remove-grapher-exports.owid.pages.dev/grapher/life-expectancy.png?imWidth=850) 500s, but other resolutions work fine

It's working fine for me right now, so this might be intermittent? Which is still not great, though.



> * Seems to fail due to there being too many thumbnails to load

Possibly, but introducing a `.related-charts__list.related-charts__list img { width: 100% }` rule will help this a bit by requiring fewer relayouts (and I think making `img loading="lazy"` work).

**Comment by rakyi** [comment]
@ikesau what's the status of this? It would make my mdim work easier if we could ship this before I start implementing dynamic thumbnails for them.

**Comment by ikesau** [comment]
Planning on reviving it start of next week 🙂👍

**Pull request closed without merging**